plugin/openassetio_manager_bal/BasicAssetLibraryInterface.py

```diff
@@ -297,3 +358,29 @@ def __handle_exception(exc, idx, error_callback):
             raise exc
 
         error_callback(idx, BatchElementError(code, msg))
+
+
+class BALEntityReferencePagerInterface(EntityReferencePagerInterface):
+    """
+    Simple implementation of a pager.
+
+    Bulk-queries all data, then splits up into cached pages ready to
+    regurgitate on demand.
+    """
+
+    def __init__(self, page_size, entity_references):
+        EntityReferencePagerInterface.__init__(self)
+        self.__page_num = 0
+        self.__pages = []
+
+        for page_start in range(0, len(entity_references), page_size):
+            self.__pages.append(entity_references[page_start : page_start + page_size])
+
+    def hasNext(self, _hostSession):
+        return self.__page_num < len(self.__pages) - 1
+
+    def next(self, _hostSession):
+        self.__page_num += 1
+
+    def get(self, _hostSession):
+        return self.__pages[self.__page_num] if self.__page_num < len(self.__pages) else []
```

Comment:
[consider] You likely considered this but discarded it due to the inconveniences of moving the method, but it would be more representative if these methods had the simulated_delay on them, rather than the pager-creation methods.

Reply:
Ahh yeah - well remembered - it was meant to be on both wasnt it.

Reply:
Yup very good point.

Reply:
Added in f3405e96 - somewhat bigger change than I expected

Reply:
:+1: